File: miband-sdk/src/main/java/com/khmelenko/lab/miband/listeners/ActionCallback.java
package com.khmelenko.lab.miband.listeners;

import android.bluetooth.BluetoothGattCharacteristic;

/**
 * Action callback
 *
 * @author Dmytro Khmelenko
 */
public interface ActionCallback {

    /**
     * Called on successful completion
     *
     * @param data Fetched data
     */
    void onSuccess(Object data);

    /**
     * Called on fail
     *
     * @param errorCode Error code
     * @param msg       Error message
     */
    void onFail(int errorCode, String msg);
}

File: miband-sdk-kotlin/src/main/java/com/khmelenko/lab/miband/listeners/ActionCallback.kt
package com.khmelenko.lab.miband.listeners

/**
 * Action callback

 * @author Dmytro Khmelenko
 */
interface ActionCallback {

    /**
     * Called on successful completion

     * @param data Fetched data
     */
    fun onSuccess(data: Any)

    /**
     * Called on fail

     * @param errorCode Error code
     * *
     * @param msg       Error message
     */
    fun onFail(errorCode: Int, msg: String)
}
